src/controllers/tripRequest.comments.js

```diff
@@ -8,15 +8,24 @@ export const commentOnRequests = async (req, res, next) => {
   try {
     const { comment } = req.body;
     const userId = req.user.id;
+    const user = await users.findByPk(userId);
     const tripRequestId = req.params.tripRequestId;
-    const tripRequest = await tripRequests.findByPk(tripRequestId);
+    const tripRequest = await tripRequests.findOne({
+      where: { requesterId: userId, id: tripRequestId },
+    });
 
     if (!tripRequest) {
       return res.status(400).json({
         error: 'The trip request you are trying to comment on does not exist!',
       });
     }
 
+    if (userId !== tripRequest.requesterId && user.role !== 'manager') {
```

Comment:
In this case no one will be able to add a comment since the role of the `requester` is not `manager` and manager is not the owner of the trip request,  use `||` instead of `&&`

Reply:
> In this case no one will be able to add a comment since the role of the `requester` is not `manager` and manager is not the owner of the trip request, use `||` instead of `&&`

Hi @eddy you are wrong,
Are you sure that no one will be able to add a comment?
Please try to test it on your computer so you be sure.

let me tell you what would happen If I use `userId !== tripRequest.requesterId || user.role !== 'manager`, 
- one of the conditions would always be true because;
-  if `userId === tripRequest.requesterId` that means the user role is requester and not manager which means the second condition is true

- if `user.role === manager` that means that the user is not a requester and the first condition would return `true`

- NOTE THIS if I use `||` and one of the conditions return true, WHICH IS THE CASE, the if statement would return true because if you are using `||` and if one of the conditions is true, then the whole condition would return `true` and the user will get this response in that if statement  `'You are Not Authorized to comment on this request'`

Reply:
fine, I get it

Reply:
Thanks for testing @uwambajeddy 
I fixed that you may test again

Reply:
Thanks for testing @uwambajeddy 
I fixed that, you may test again

Reply:
🤝

Reply:
Fixed that.
Thanks!.